### PR TITLE
Added batching option

### PR DESCRIPTION
**Pull request by HtheChemist**

### CODE REVIEW

File: valexa/gui/app.py [hunk 1]
```diff
@@ -98,3 +117,33 @@ def on_tolerance_changed(self, text):
     def on_acceptance_changed(self, text):
         if text:
             self.state.acceptance_limit = int(text)
+
+    def passive_mode(self):
```
Comment (kimond) [inline valexa/gui/app.py]: `passive_mode` is not very meaningful as a function name.

File: valexa/gui/profile_widget.py [hunk 2]
```diff
@@ -34,19 +34,49 @@ def __init__(self, profile: Profile):
...
+            table.setItem(index + 2, 3, QTableWidgetItem(f'{list_item.relative_bias:.2f}'))
+            table.setItem(index + 2, 4, QTableWidgetItem(f'{list_item.repeatability_std_pc:.3f}'))
+            table.setItem(index + 2, 5, QTableWidgetItem(f'{list_item.inter_series_std_pc:.3f}'))
+            #table.setItem(index+1, 4, QTableWidgetItem(f'{list_item.recovery:.1f}'))
+            table.setItem(index + 2, 6,
+                          QTableWidgetItem(f'[{list_item.abs_tolerance[0]:.2f},{list_item.abs_tolerance[1]:.2f}]'))
+            table.setItem(index + 2, 7,
+                          QTableWidgetItem(f'[{-100 + list_item.rel_tolerance[0]:.2f},{-100 + list_item.rel_tolerance[1]:.2f}]'))
+
+        #table.setHorizontalHeaderLabels(
```
Comment (kimond) [inline valexa/gui/profile_widget.py]: If this block of code is no more used it should not remain in the code as a smelly comment.

File: valexa/gui/profile_widget.py [hunk 3]
```diff
@@ -34,19 +34,49 @@ def __init__(self, profile: Profile):
...
+
+
+        for (index, list_item) in enumerate(self.profile.levels):
+            table.setItem(index + 2, 0, QTableWidgetItem(f'{list_item.introduced_concentration:.3f}'))
+            table.setItem(index + 2, 1, QTableWidgetItem(f'{list_item.calculated_concentration:.3f}'))
+            table.setItem(index + 2, 2, QTableWidgetItem(f'{list_item.bias:.3f}'))
+            table.setItem(index + 2, 3, QTableWidgetItem(f'{list_item.relative_bias:.2f}'))
+            table.setItem(index + 2, 4, QTableWidgetItem(f'{list_item.repeatability_std_pc:.3f}'))
+            table.setItem(index + 2, 5, QTableWidgetItem(f'{list_item.inter_series_std_pc:.3f}'))
+            #table.setItem(index+1, 4, QTableWidgetItem(f'{list_item.recovery:.1f}'))
```
Comment (kimond) [inline valexa/gui/profile_widget.py]: Smelly comment. if not more used remove it.

File: valexa/gui/app.py [hunk 4]
```diff
@@ -60,36 +64,51 @@ def open_data_page(self):
...
-                                     self.state.acceptance_limit)
-            plot_layout = self.plot_layout
-            for profile in profiles:
-                profile_widget = ProfileWidget(profile=profile)
-                plot_layout.addWidget(profile_widget)
-        except Exception as e:
-            QMessageBox.critical(self, "Error", str(e))
-
-    def select_xlsx(self):
+        #try:
```
Comment (kimond) [inline valexa/gui/app.py]: smelly comment, should be remove if not more used

File: valexa/gui/app.py [hunk 5]
```diff
@@ -60,36 +64,51 @@ def open_data_page(self):
...
+        print('try')
+        profiles = make_profiles(self.state.calib_data, self.state.valid_data, self.state.tolerance_limit,
+                                 self.state.acceptance_limit, PurePath(self.file_name_input.text()).name)
+        plot_layout = self.plot_layout
+        for profile in profiles:
+
+            profile_widget = ProfileWidget(profile=profile)
+            plot_layout.addWidget(profile_widget)
+
+        #except Exception as e:
```
Comment (kimond) [inline valexa/gui/app.py]: smelly comment here again

File: valexa/core/xlsx.py [hunk 6]
```diff
@@ -43,3 +51,131 @@ def __get_data(self, sheet_name):
...
+
+    def __valid_if_file_exist(self, file_name):
+        return path.exists(file_name)
+
+class HtmlWriter:
+
+    def __init__(self, profile_to_use):
+        self.profile = profile_to_use
+
+    def write_profile(self, file_name):
```
Comment (kimond) [inline valexa/core/xlsx.py]: the following logic seems to be linked to the HTML exportation. Should we move this outside the xlsx module? I think so.

File: valexa/core/profiles.py [hunk 7]
```diff
@@ -8,15 +8,17 @@
...
 from valexa.core.models import Result, ModelHandler, Model
 
 DEFAULT_TOLERANCE = 80
-DEFAULT_ACCEPTANCE = 20
+DEFAULT_ACCEPTANCE = 50
 
 
 def make_profiles(calib_data: List[tuple], valid_data: List[tuple], tolerance_limit: int,
-                  acceptance_limit: int) -> List:
+                  acceptance_limit: int, file_name: str) -> List:
```
Comment (kimond) [inline valexa/core/profiles.py]: What is the purpose of the filename here?

File: valexa/gui/app.py [hunk 8]
```diff
@@ -60,36 +64,51 @@ def open_data_page(self):
         self.stackedWidget.setCurrentIndex(1)
 
     def open_model_page(self):
+
```
Comment (kimond) [inline valexa/gui/app.py]: remove empty line between function definition and the block of code

File: valexa/gui/app.py [hunk 9]
```diff
@@ -60,36 +64,51 @@ def open_data_page(self):
...
-                self.validation_data_list_widget.addItems([str(r) for r in self.state.valid_data])
+                elif os.path.splitext(filename)[1] == ".xlsx":
+                    self.select_xlsx(filename)
 
         except InvalidFileException as e:
-            QMessageBox.critical(self, "Error", str(e))
+            QMessageBox.critical(self, "Error 2", str(e))
+
+    def select_xlsx(self, filename):
+
```
Comment (kimond) [inline valexa/gui/app.py]: no blank line here

File: valexa/gui/profile_widget.py [hunk 10]
```diff
@@ -34,19 +34,49 @@ def __init__(self, profile: Profile):
...
+        table.setItem(1, 2, QTableWidgetItem('Biais absolue'))
+        table.setItem(1, 3, QTableWidgetItem('Biais relatif'))
+
+        table.setItem(1, 4, QTableWidgetItem('Répétablité (%RSD)'))
+        table.setItem(1, 5, QTableWidgetItem('Précision intermédaire (%RSD)'))
+
+        table.setItem(1, 6, QTableWidgetItem('Intervales de tolérance'))
+        table.setItem(1, 7, QTableWidgetItem('Limite de tolérance relative'))
+
+
```
Comment (kimond) [inline valexa/gui/profile_widget.py]: max 1 blank line

File: valexa/core/profiles.py (hunk 7 above)
Comment (HtheChemist) [reply valexa/core/profiles.py]: file_name is used in the creation of the report where it put the file_name in the header of the graph. See core/html.py line 30.

File: .idea/misc.xml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kimond) [inline .idea/misc.xml]: this file should be ignores thus not commited

File: .idea/vcs.xml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kimond) [inline .idea/vcs.xml]: this file should be ignored thus not commited

**Comment by kimond** [comment]
@HtheChemist some test fails but I could fix them later

**Pull request merged**